Feat/redux

a-nester opened the pull request.

a-nester committed:
add/redux

a-nester committed:
Merge branch 'main' into feat/redux

a-nester committed:
action/login-proceed

a-nester committed:
Merge branch 'main' into feat/redux

a-nester committed:
add/redux-auth/App/Priv-and-Restricted-routes

a-nester committed:
fix/current-fetch

gudzsv commented on src/helpers/axios.js:
```diff
@@ -0,0 +1,29 @@
+    if (error.response.status === 401 && !originalRequest._retry) {
+      originalRequest._retry = true;
+      try {
+        store.dispatch(refreshUser());
```
please use await 
like this 
        await store.dispatch(refreshUser()).unwrap();
        await store.dispatch(currentUser()).unwrap();

gudzsv commented on src/helpers/axios.js:
```diff
@@ -0,0 +1,29 @@
+import { store } from 'redux/store';
+
+export const API = axios.create({
+  baseURL: '',
```
add to baseURL value from constant or env server link: https://h2o-experts-server.onrender.com

gudzsv commented on src/App.jsx:
```diff
@@ -13,35 +17,34 @@ const NotFoundPage = lazy(() =>
+  return isRefreshing ? (
+    <Loader />
+  ) : (
+    <>
```
please remove fragment <>

gudzsv commented on src/App.jsx:
```diff
@@ -1,8 +1,12 @@
+import { lazy, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import { Routes, Route } from 'react-router';
+import { refreshUser } from 'redux/auth/operations.js';
```
Please check all path to imported file

gudzsv commented on src/helpers/axios.js:
```diff
@@ -0,0 +1,29 @@
+import axios from 'axios';
+import { currentUser, refreshUser } from 'redux/auth/operations';
```
Please check path to all imported files

gudzsv commented on src/helpers/axios.js:
```diff
@@ -0,0 +1,29 @@
+        store.dispatch(currentUser());
+      } catch (refreshError) {
+        // Handle refresh token errors by clearing stored tokens and redirecting to the login page.
+        console.error('Token refresh failed:', refreshError);
```
Please remove console.log

gudzsv commented on src/redux/auth/operations.js:
```diff
@@ -1,83 +1,83 @@
+  'auth/register',
+  async (registerData, thunkAPI) => {
+    try {
+      await API.post('/users/signup', registerData);
```
I believe that endpoint shoul be /user/register

gudzsv commented on src/redux/auth/operations.js:
```diff
@@ -1,83 +1,83 @@
+  async (registerData, thunkAPI) => {
+    try {
+      await API.post('/users/signup', registerData);
+      const { data } = await API.post('/users/signin', registerData);
```
I believe that endpoint shoul be /user/login

gudzsv commented on src/redux/auth/operations.js:
```diff
@@ -1,83 +1,83 @@
+  'auth/login',
+  async (loginData, thunkAPI) => {
+    try {
+      const { data } = await API.post('/users/login', loginData);
```
please change all /users to /user

gudzsv commented on src/redux/auth/operations.js:
```diff
@@ -1,83 +1,83 @@
-		const response = await axios.post('/users/logout');
-		clearAuthHeader();
+  try {
+    const { data } = await API.post('/users/logout');
```
please change users to user

gudzsv commented on src/redux/auth/operations.js:
```diff
@@ -1,83 +1,83 @@
+  'auth/current',
+  async (_, thunkAPI) => {
+    try {
+      const response = await API.get('/users/');
```
please change users to user

gudzsv commented on src/redux/auth/operations.js:
```diff
@@ -1,83 +1,83 @@
+  'auth/refresh',
+  async (_, thunkAPI) => {
+    try {
+      const { data } = await API.get('/users/refresh');
```
please change users to user

gudzsv commented on src/redux/store.js:
```diff
@@ -9,8 +10,8 @@ const placeholderReducer = (state = {}, action) => {
 export const store = configureStore({
```
please continue configure store 
and check if aplecation run after all changes

gudzsv reviewed (changes requested):
Good job Thanks 
I have some recomendation for you 
Please read my recommendations and apply the changes.
Also, before you send the changes, make sure that the application runs without errors. Because this will block others from working with the project.
Thanks again for the work done. Let's not stop, let's move on.

a-nester committed:
fix/path/add/patch

a-nester committed:
Merge branch 'main' into feat/redux

a-nester committed:
add/persist

a-nester committed:
Merge branch 'main' into feat/redux

a-nester committed:
update-branch

gudzsv committed:
Changed signup and signin on register and login

     await API.post('/user/signup', registerData);
      const { data } = await API.post('/user/signin', registerData);

on

      await API.post('/users/register', registerData);
      const { data } = await API.post('/users/login', registerData);

Co-Authored-By: Alexandr Nesterenko <73670421+a-nester@users.noreply.github.com>

gudzsv reviewed (approved):
@a-nester Excellent work. thank you very much for the fast implementation of redux
Approve
